### PR TITLE
Support parsing resolved InetSocketAddress string

### PR DESCRIPTION
**Pull request by YangchenYe323**
### What changes are proposed in this pull request?
Modify `NetworkAddressUtils#parseInetSocketAddress` to handle resolved address strings correctly.

### Why are the changes needed?
As mentioned in https://github.com/Alluxio/alluxio/pull/15670#discussion_r891870760 A resolved `InetSocketAddress` has the following `toString()` representation: "<hostname>/<resolved_address>:<port>". It is handy that `NetworkAddressUtils#parseInetSocketAddress` can process such strings correctly, so that we can pass an `InetSocketAddress` directly to the configuration.

### Does this PR introduce any user facing changes?
No

### CODE REVIEW

**Comment by YangchenYe323** [comment]
@dbw9580 Do you mind reviewing this? Thanks!

File: core/server/worker/src/test/java/alluxio/worker/block/BlockMasterClientTest.java [hunk 1]
```diff
@@ -77,17 +77,14 @@
...
-  // This string cannot be parsed accurately by the current parsing algorithm
+  // test socket address
   private static final InetSocketAddress TEST_SOCKET_ADDRESS =
       new InetSocketAddress("localhost", 9999);
-  private static final String TEST_SOCKET_ADDRESS_STRING = "localhost:9999";
 
   @Rule
   public ConfigurationRule mConfiguration = new ConfigurationRule(ImmutableMap
       .of(
-          PropertyKey.MASTER_RPC_ADDRESSES, ImmutableList.of(TEST_SOCKET_ADDRESS_STRING),
```
Comment (YangchenYe323) [inline core/server/worker/src/test/java/alluxio/worker/block/BlockMasterClientTest.java:90]: After this fix, we can use `InetSocketAddress` directly

File: core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java [hunk 2]
```diff
@@ -666,7 +666,11 @@ public static InetSocketAddress parseInetSocketAddress(String address) throws IO
     if (strArr.length != 2) {
       throw new IOException("Invalid InetSocketAddress " + address);
     }
-    return InetSocketAddress.createUnresolved(strArr[0], Integer.parseInt(strArr[1]));
+    // a typical resolved InetSocketAddress will have hostname part <hostname>/<address>,
```
Comment (dbw9580) [inline core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java:669]:
```suggestion
    // a typical resolved InetSocketAddress has a string representation of form <hostname>/<address>:port,
```

File: core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java [hunk 3]
```diff
@@ -666,7 +666,11 @@ public static InetSocketAddress parseInetSocketAddress(String address) throws IO
...
+    // e.g., "localhost/127.0.0.1:9000".
+    // extract the <hostname> part by splitting at "/"
+    String[] hostname = strArr[0].split("/");
+    return InetSocketAddress.createUnresolved(hostname[0], Integer.parseInt(strArr[1]));
```
Comment (dbw9580) [inline core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java:673]:
```suggestion
    String hostname = strArr[0].split("/")[0];
    return InetSocketAddress.createUnresolved(hostname, Integer.parseInt(strArr[1]));
```

**Review by dbw9580 — APPROVED** [review]
LGTM!

**Comment by YangchenYe323** [comment]
@jiacheliu3 Could we merge this change? Thanks!

**Comment by jiacheliu3** [comment]
@YangchenYe323 This one is marked CLA missing, did you sign the CLA?

**Comment by YangchenYe323** [comment]
> @YangchenYe323 This one is marked CLA missing, did you sign the CLA?

Yes I did. My previous PRs didn't have the label. Does this prevent merging?

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Comment by jiacheliu3** [comment]
@YangchenYe323 Yea the merge will be rejected. Do you know what went wrong?

**Comment by YangchenYe323** [comment]
> @YangchenYe323 Yea the merge will be rejected. Do you know what went wrong?

I'll take a look.

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**